backoff/_wait_gen.py

```diff
@@ -21,12 +21,11 @@ def expo(
     """
     # Advance past initial .send() call
     yield  # type: ignore[misc]
-    n = 0
+    a = 1
     while True:
-        a = factor * base ** n
         if max_value is None or a < max_value:
```

Comment:
This `max_value` condition doesn't work properly with your changes, it needs to be updated a bit, like this, but still 24x faster than the original:

```python
    base_n = 1
    while True:
        a = factor * base_n
        if max_value is None or a < max_value:
            yield a
            base_n *= base
```